lib/hawkular/metrics/metrics_client.rb

```diff
@@ -43,6 +46,9 @@ def initialize(entrypoint,
                    options = {})
       entrypoint = normalize_entrypoint_url entrypoint, 'hawkular/metrics'
       super(entrypoint, credentials, options)
+      version = fetch_version_and_status['Implementation-Version']
+      major, minor = version.scan(/\d+/).map(&:to_i)
```

Comment:
if fetch_version_and_status fails, it does mean that your entry point is incorrect because as far as I understand if the service is up this shouldn't happen.

So, if the version/status is corrupted we can raise an exception here.

WDYT? @pilhuhn